src/jsMain/js/Simulator/StatusPanel/index.jsx

```diff
@@ -0,0 +1,105 @@
+import React, { Fragment } from 'react';
+import FakeClientDevice from '../FakeClientDevice';
+
+// todo: real interface with WebUi and Mapper implementors
+// interface HostedWebApp {
+//     fun render(node);
+//     fun onResize(node);
+//     fun onClose(node);
+// }
+
+let nextHostedWebAppId = 0;
+let hostedWebApps = {};
+
+class StatusPanel extends React.Component {
+  state = {
+    hostedWebAppIds: [],
+    hostedWebApps: {},
+  };
+
+  handleWebUiClick = () => {
+    const thisId = nextHostedWebAppId++;
+
+    const webUi = {
+      id: thisId,
+      // app: new WebUI(),
+
+      onClose: function() {
+        console.log('closed!');
+      },
+      render: function(node) {
+        // Question: What is the render function for?
```

Comment:
I'm not entirely sure what this render function is for? Probably also should have asked this as a GitHub comment instead of committing a comment in the file, but see question below in the code

Reply:
That's where the hosted web app gets to render its own content… `render: (node) => node.innerText = 'stuff!'`

Reply:
Think of it as roughly equivalent to providing `document.body`.